docs/src/main/asciidoc/config.adoc

```diff
@@ -131,6 +131,31 @@ public String hello() {
 TIP: Use `@io.smallrye.config.ConfigMapping` annotation to group multiple configurations in a single interface. Please,
 check the https://smallrye.io/smallrye-config/Main/config/mappings/[Config Mappings] documentation.
 
+== Store secrets in local configuration
+
+A secret (such as a password, a personal access token or an API key) must not end up in version control
+for security reasons. One way is to store them in a local configuration file:
+
+. Store the secret in the `.env` file in the project root directory.
++
+.The .env file
+[source,properties]
+----
+FOO_API_KEY=ThisIsSecret
+----
+
+. Add the `.env` file to `.gitignore`.
+
+. Load the environment variable as a property.
++
+.application.properties
+[source,properties]
+----
+foo.api-key=${FOO_API_KEY}
+----
```

Comment:
This is an optional step, given that Quarkus already registers the `.env` file as a ConfigSource automatically. You can also use lowercase keys in the `.env` file

Reply:
Do you mean the `foo.api-key` entry isn't needed for it to work?

Reply:
Fixing like this:
![image](https://github.com/quarkusio/quarkus/assets/176880/c608cdda-00b5-48c1-9334-ebf96a28dd4f)